modules/bridge/src/main/ts/ephox/bridge/core/MenuButton.ts

```diff
@@ -6,19 +6,28 @@ import { NestedMenuItemContents } from '../components/menu/NestedMenuItem';
 export type MenuButtonItemTypes = NestedMenuItemContents;
 export type SuccessCallback = (menu: string | MenuButtonItemTypes[]) => void;
 
+// NOTE: FetchContext is an object so that we can add information to it in the future.
+export interface FetchContext {
+  pattern: string;
+}
+
 export interface BaseMenuButtonSpec {
   text?: string;
   tooltip?: string;
   icon?: string;
-  fetch: (success: SuccessCallback) => void;
+  searchable?: boolean;
+  // In order to avoid breaking APIs with pre 6.2 releases, the fetchContext was added
+  // as an additional argument to fetch.
+  fetch: (success: SuccessCallback, fetchContext?: FetchContext) => void;
```

Comment:
```suggestion
  fetch: (success: SuccessCallback, fetchContext: FetchContext) => void;
```
Adding new arguments isn't breaking in the JS/TS ecosystem for something like this, as if someone only gives a function with just he success then it'll work fine still. Now if the integrators had to call this function it would be breaking, but it's us who are doing that. In addition making this optional would just make it hard for integrators since they'd have to add checks to make sure the fetch context is not undefined.

Reply:
Ah, of course. They can define a function without the full number of arguments if they want. Yeah, I'm used to that being an incompatible type signature. Don't try that in haskell...

Reply:
I should add if we actually aren't always passing an argument (which I didn't check) then this would have to be optional 🤷